ironic/provider.go

```diff
@@ -90,8 +132,52 @@ func configureProvider(schema *schema.ResourceData) (interface{}, error) {
 		if err != nil {
 			return nil, fmt.Errorf("could not configure inspector endpoint: %s", err.Error())
 		}
-		clients.Inspector = inspector
+		clients.inspector = inspector
+	}
+
+	clients.timeout = time.Duration(schema.Get("timeout").(int)) * time.Second
+
+	return &clients, err
+}
+
+// Retries a gophercloud API until a timeout is reached.
+func waitForAPI(duration time.Duration, client *gophercloud.ServiceClient) error {
+	success := make(chan bool, 1)
+	timeout := make(chan bool, 1)
+	defer close(success)
+	defer close(timeout)
+
+	go func() {
+		for tries := 0; ; tries++ {
```

Comment:
Would this code be simpler and more readable if we just calculated the maximum number of tries for a given timeout?

Obviously we'd need to ensuire the built in httpclient instance also has a sensible timeout to handle the case where the Get hangs up.

Reply:
The time.After built-in channel seems to be how a lot of people do timeouts in Go: https://gobyexample.com/timeouts

But, I could do it that way.  I don't think I could measure the length of the GET request itself, so the timeout could be off by a bit.

Reply:
Fair enough - I was thinking we'd set the GET timeout to a known value but if this is the common pattern then I'm not blocking on it :)

Reply:
Ok, I'd prefer to keep it as is as my bandwidth to take on more things is limited trying to shepherd the openshift/installer PR through. If the goroutine/`time.After` thing becomes an issue I can refactor it to do it the other way.